src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -62,6 +62,66 @@ size_t AdaptDispatch::_SetRgbColorsHelper(const VTParameters options,
     return optionsConsumed;
 }
 
+// Routine Description:
+// - Helper to parse extended graphics options, which start with 38 (FG) or 48 (BG)
+//   - These options are followed by either a 2 (RGB) or 5 (xterm index):
+//     - RGB sequences then take 4 MORE options to designate the ColorSpaceID, R, G, B parts
+//       of the color.
+//     - Xterm index will use the option that follows to use a color from the
+//       preset 256 color xterm color table.
+// Arguments:
+// - colorItem - One of FG(38) and BG(48), indicating which color we're setting.
+// - options - An array of options that will be used to generate the RGB color
+// - attr - The attribute that will be updated with the parsed color.
+// Return Value:
+// - <none>
+void AdaptDispatch::_SetRgbColorsHelperAlt(const VTParameter colorItem,
+                                           const VTSubParameters options,
+                                           TextAttribute& attr) noexcept
+{
+    // This should be called for applying FG and BG colors only.
+    assert(colorItem == GraphicsOptions::ForegroundExtended ||
+           colorItem == GraphicsOptions::BackgroundExtended);
+
+    const bool isForeground = (colorItem == GraphicsOptions::ForegroundExtended);
+    const DispatchTypes::GraphicsOptions typeOpt = options.at(0);
+
+    if (typeOpt == DispatchTypes::GraphicsOptions::RGBColorOrFaint)
+    {
+        // sub params are in the order:
+        // :2:<color-space-id>:<r>:<g>:<b>
+        // we skip color-space-id, because we don't support it.
+        const size_t red = options.at(2).value_or(0);
+        const size_t green = options.at(3).value_or(0);
+        const size_t blue = options.at(4).value_or(0);
+        // ensure that each value fits in a byte
+        if (red <= 255 && green <= 255 && blue <= 255)

```

Comment:
How do other terminals handle this? Should we maybe just apply `std::min(red, 255)` to the parameters here, so that slightly invalid sequences aren't discarded?

Reply:
Again this is a mirror of the implementation in `_SetRgbColorsHelper`, and it was done this way to match XTerm and VTE. Without a reference implementation to work from, following their lead seemed like the next best thing.

Reply:
I just used what I got from the other helper `_SetRgbColorsHelper`.

Reply:
> Without a reference implementation to work from, following their lead seemed like the next best thing.

Can you suggest what we can improve here? Though, I don't see we need anychanges here, but I'm open to it.

Reply:
As far as I'm concerned, it's fine as it is. If there was a hardware implementation, I'd say we should be matching the original hardware, but there's isn't any. If there was something in the spec that defined the behavior of out of range parameters, I'd say we should follow that, but I couldn't find anything there either.

So the next best thing is to follow whatever everyone else is doing. And in this case there's quite a lot of variation amongst the different terminals, but at least XTerm and VTE were in agreement that colors with out of range parameters should be ignored (or so it appeared in my testing).

If we were the first ones to implement this, I would probably have gone with clamping, as @lhecker suggested, but I think compatibility with XTerm and VTE is of more value.

Reply:
Ah, thanks for the explanation! I wasn't aware about this. We should probably make sure to add this as a code comment at some point.
@tusharsnn If you don't mind, could you maybe replace the "ensure that each value fits in a byte" comment with something like "Rejecting out of range values match XTerm's and VTE's behavior" or similar? I think that would be really helpful for future readers!

Reply:
Yeah, sorry, that's on me. I'm terrible with comments. I definitely should have made a note about this.

Reply:
@lhecker Sure, will update the comment.